projects/packages/connection/src/class-webhooks.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Connection Webhooks class.
+ *
+ * @package automattic/jetpack-connection
+ */
+
+namespace Automattic\Jetpack\Connection;
+
+use Automattic\Jetpack\Roles;
+use Automattic\Jetpack\Tracking;
+use Jetpack_Options;
+
+/**
+ * Connection Webhooks class.
+ */
+class Webhooks {
+
+	/**
+	 * The Connection Manager object.
+	 *
+	 * @var Manager
+	 */
+	private $connection;
+
+	/**
+	 * Webhooks constructor.
+	 *
+	 * @param Manager $connection The Connection Manager object.
+	 */
+	public function __construct( $connection ) {
+		$this->connection = $connection;
+	}
+
+	/**
+	 * Initialize the webhooks.
+	 *
+	 * @param Manager $connection The Connection Manager object.
+	 */
+	public static function init( $connection ) {
+		$webhooks = new static( $connection );
+
+		add_action( 'init', array( $webhooks, 'controller' ) );
+	}
+
+	/**
+	 * The "controller" decides which handler we need to run.
+	 */
+	public function controller() {
```

Comment:
Do you think unit tests for `Webhooks::controller()` would provide any value?

Reply:
Completed, thanks for pointing that out.